EIPS/eip-draft_estimate_gas_return_data.md

```diff
@@ -0,0 +1,39 @@
+---
+eip: _
+title: Estimate gas return data
+author: Jules Goullée (@julesGoullee)
+discussions-to: _
+status: Draft
+type: Standards Track
+category: Interface
+created: 2020-06-15
+---
+
+## Abstract
+This EIP intends to modify RPC to return transaction data for `eth_estimateGas` method. When transaction required state modification, (not a Solidity view or pure, it can't be executed without a transaction), the estimate endpoint returns the amount of gas needed but omit the returned value. The propose of this change is to add the return value in the response.
+
+## Motivation
+Pure of view function can be call without cost to "read" information and free of cost. State modifier function can be estimate to know the cost and if it will not fail, but there is no way to know the return value before executing the transaction in live. 
```

Comment:
This isn't true currently with any clients I'm familiar with.  You can always do an `eth_call` of any transaction that you would have done an `eth_estimateGas` or `eth_sendTransaction` for.  The only motivation I can see here is that instead of having to do an `eth_call` *and* an `eth_estimateGas`, you can instead just do one of them.

That being said, getting the result of an on-chain transaction like this is *generally* a bad practice since the transaction may not execute the same on-chain as it does off-chain, so I'm generally against a proposal to encourage that behavior.  Really, contracts should be emitting events with results, or we need to include results in transaction receipts.